CONVENTIONS.md

```diff
@@ -141,20 +141,129 @@ it is used to manage all elements of the cluster.
-then the component should request `100m/600m * 350m`).  Components
-scheduled to all nodes should be proportional to the SDN CPU request
-during the standard e2e suite.
-
-The memory request of cluster components should be set at 10% higher than their p90 memory usage over a standard e2e suite execution.
-
-Limits should be not set because components cannot anticipate how they scale in usage in all customer environments and a crashlooping OOM component is something we must detect and handle gracefully regardless. Setting memory limits leaves administrators with no option to react to valid changes usage dynamically.
-
+Kubernetes allows Pods to declare their CPU and memory resource
+requirements in advance using [requests and
+limits](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/). Requests
+are used to ensure minimum resources are provided and to influence
+scheduling, while limits prevent Pods from consuming resources
+excessively.
+
+Unlike with user workloads, setting limits for cluster components is
+problematic for several reasons:
+
+* Components cannot anticipate how they scale in usage in all customer
+  environments, so setting one-size-fits-all limits is not possible.
+* Setting static limits prevents administrators from responding to
+  changes in their clusters’ needs, such as by resizing control plane
+  nodes to provide more resources.
+* We do not want cluster components to be restarted based on their
+  resource consumption (for example, being killed due to an
+  out-of-memory condition). We need to detect and handle those cases
+  more gracefully, without degrading cluster performance.
+
+Therefore, cluster components *SHOULD NOT* be configured with resource
+limits.
+
+However, cluster components *MUST* declare resource requests for both
+CPU and memory.
+
+Specifying requests without limits allows us to ensure that all
+components receive their required minimum resources and that they are
+able to burst to use more resources as needed. When setting the
+resource requests, we need to balance the size between values that
+ensure the component has the resources it needs to keep running with
+the requirement to be efficient and support small-footprint
+deployments. If the request settings for a component are too low, it
+could be starved for resources in a very busy cluster, resulting in
+degraded performance and lower quality of service. If the request
+settings for a component are too high, it could mean that the
+scheduler cannot find a place to run it, leading to crash looping or
+preventing a cluster from deploying. If the combined resource requests
+for the cluster components are too high, users may not have enough
+resources available to use OpenShift at all, resulting in lost
+business (see especially the [single-node](enhancements/single-node)
+use cases).
+
+Kubernetes resource requests for CPU time are measured in fractional
```

Comment:
I think having a paragraph before this to distinguish and set tone for future use cases is important - distinguishing between best practice for setting requests on compressible resources (CPU) and best practices on incompressible resources (memory).  For incompressible we wish to set a request that is roughly aligned to what we will use.  For compressible resources we are more concerned about balancing between proportional users - ensuring system or user workloads are not unfairly preferred (if system compressible requests are too low, user workloads will take priority and lead to impact to the user workload, vice versa some user workloads will not schedule) as well as ensuring that in resource constrained situations there is enough proportional CPU for the system to make progress.

Reply:
I don't fully understand what you mean by "compressible" and "incompressible". Can you explain those terms?

Reply:
CPU is compressible because if there's less of it you generally just go slower, storage resources (RAM, disk) are incomprehensible because when you run out of those you halt.

https://en.wikipedia.org/wiki/System_resource#Categories

Reply:
TIL, thanks @sdodson 

I've added a paragraph combining what you both said at line 187 in the latest draft.